core/src/main/java/feast/core/service/SpecService.java

```diff
@@ -196,6 +204,129 @@ public ListFeatureSetsResponse listFeatureSets(ListFeatureSetsRequest.Filter fil
     return response.build();
   }
 
+  /**
+   * Return a map of feature references and features matching the project, labels and entities
+   * provided in the filter. All fields are required.
+   *
+   * <p>Project name can be explicitly provided, or an asterisk can be provided to match all
+   * projects. A combination of asterisks/wildcards and text can be used to find features across
+   * multiple matching projects. If the project name is omitted, the default project would be used.
+   *
+   * <p>The entities in the filter accepts a list. All matching features will be returned. Regex is
+   * not supported. If no entities are provided, features will not be filtered by entities.
+   *
+   * <p>The labels in the filter accepts a map. All matching features will be returned. Regex is not
+   * supported. If no labels are provided, features will not be filtered by labels.
+   *
+   * @param filter filter containing the desired project name, entities and labels
+   * @return ListEntitiesResponse with map of feature references and features found matching the
+   *     filter
+   */
+  public ListFeaturesResponse listFeatures(ListFeaturesRequest.Filter filter)
+      throws InvalidProtocolBufferException {
+    String project = filter.getProject();
+    List<String> entities = filter.getEntitiesList();
+    Map<String, String> labels = filter.getLabelsMap();
+
+    checkValidCharactersAllowAsterisk(project, "projectName");
+
+    // Autofill default project if project not specified
+    if (project.isEmpty()) {
+      project = Project.DEFAULT_NAME;
+    }
+
+    // Currently defaults to all FeatureSets
+    List<FeatureSet> featureSets;
+
+    if (project.contains("*")) {
```

Comment:
I think we shouldnt support multiple projects here for now, and also not wildcards. Can we just limit it to a single project for the time being?